### PR TITLE
Working towards a smaller FormEntryActivity

### PR DESCRIPTION
**Pull request by phillipm**
The `FormEntryActivity` class is 1800 lines long, this PR starts breaking it up into several different classes to make it more readable.

### CODE REVIEW

File: app/src/org/commcare/activities/components/FormEntrySessionWrapper.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import static org.commcare.activities.FormEntryActivity.mFormController;
+
+/**
+ * Wrapper for the form entry action recorder, allowing for saving and
+ * restoring to the form entry activity.
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class FormEntrySessionWrapper {
```
Comment (amstone326) [inline app/src/org/commcare/activities/components/FormEntrySessionWrapper.java]: What's the reason this code can't just be part of `FormEntrySession`?

File: app/src/org/commcare/activities/components/FormEntryInstanceUtils.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import org.commcare.provider.FormsProviderAPI;
+import org.commcare.provider.InstanceProviderAPI;
+import org.commcare.utils.StringUtils;
+import org.commcare.views.dialogs.DialogChoiceItem;
+import org.commcare.views.dialogs.PaneledChoiceDialog;
+
+/**
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+public class FormEntryInstanceUtils {
```
Comment (amstone326) [inline app/src/org/commcare/activities/components/FormEntryInstanceUtils.java]: What do you think about putting all of these methods into `FormEntryInstanceState.java`? I feel like there is a balance to strike here between making mammoth classes smaller and having an unnecessary number of small classes that don't have a clear reason for being separate entities and make it kind of cumbersome to navigate through code.

File: app/src/org/commcare/activities/components/FormEntrySessionWrapper.java (hunk 1 above)
Comment (phillipm) [inline app/src/org/commcare/activities/components/FormEntrySessionWrapper.java]: It contains Android specific code and `FormEntrySession` needs to live in `commcare-core`

File: app/src/org/commcare/activities/components/FormEntryInstanceUtils.java (hunk 2 above)
Comment (phillipm) [inline app/src/org/commcare/activities/components/FormEntryInstanceUtils.java]: 👍

**Comment by phillipm** [comment]
@amstone326 I've updated the PR with your suggested changes

**Comment by amstone326** [comment]
@damagatchi retest this please

**Pull request merged**